### ISSUE TITLE
Max width option

### ISSUE THREAD

**Issue by waylaidwanderer** [sindresorhus/boxen#83]
It would be nice to have a max width option as an alternative to a set width.

**Comment by PaTTeeL** [comment]
> It would be nice to have a max width option as an alternative to a set width.

Yes, boxen is awesome. I'm using those settings to beautify the text and make them easy to copy and paste.
```
    const borderStyle = {
        topLeft: ' ',
        topRight: ' ',
        bottomLeft: ' ',
        bottomRight: ' ',
        horizontal: '─',
        vertical: ' ',
    };
    const padding = {
        left: 0,
        right: 0,
    };
    options = { ...options, borderStyle, padding };
```
But I noticed there is a space at the beginning of the top. How can I remove it? The text box gone when I'm setting 'letf: -1'.
I would appreciate it if boxen could adjust the spacing on the right and left sides of the line.

**Comment by Caesarovich** [comment]
@waylaidwanderer What would you say about `width` could be either a number or a callback function that can determine a width. Like: ```js
options.width = (availableWidth) => {
  return Math.max(30, availableWidth)
}
```
This could be a way to set a max width of 30.

**Comment by waylaidwanderer** [comment]
That looks like a decent flexible solution.
